testing/remote_signer_test/src/utils.rs

```diff
@@ -29,11 +27,82 @@ pub fn get_address(client: &Client) -> String {
     format!("http://{}:{}", ip, listening_address.port())
 }
 
-pub fn set_permissions(path: &Path, perm_octal: u32) {
-    let metadata = fs::metadata(path).unwrap();
-    let mut permissions = metadata.permissions();
-    permissions.set_mode(perm_octal);
-    fs::set_permissions(path, permissions).unwrap();
+pub fn restrict_permissions(path: &Path) {
+    #[cfg(unix)]
+    {
+        use std::fs;
+        use std::os::unix::fs::PermissionsExt;
+
+        let metadata = fs::metadata(path).unwrap();
+        let mut permissions = metadata.permissions();
+        permissions.set_mode(0o0311); // set to '*-wx--x--x'
+        fs::set_permissions(path, permissions).unwrap();
+    }
+
+    #[cfg(windows)]
+    {
+        use winapi::um::winnt::PSID;
+        use windows_acl::acl::{AceType, ACL};
+
+        let path_str = path.to_str().unwrap();
+        let mut acl = ACL::from_file_path(&path_str, false).unwrap();
+
+        let owner_sid = windows_acl::helper::string_to_sid("S-1-3-4").unwrap();
+        let entries = acl.all().unwrap();
+        // remove all AccessAllow entries
+        for entry in &entries {
+            if let Some(ref entry_sid) = entry.sid {
+                acl.remove(
+                    (*entry_sid).as_ptr() as PSID,
+                    Some(AceType::AccessAllow),
+                    None,
+                )
+                .unwrap();
+            }
+        }
+        // add single entry for minimal access to file owner
+        // allowing them only to read attributes of the file
+        // and read/modify permissions
+        acl.add_entry(
+            owner_sid.as_ptr() as PSID,
+            AceType::AccessAllow,
+            0,
+            0x160088,
```

Comment:
As per the other permissions code (use constants)

Reply:
done!